grpc-server/src/main/scala/com/avast/sst/grpc/server/interceptor/LoggingServerInterceptor.scala

```diff
@@ -0,0 +1,48 @@
+package com.avast.sst.grpc.server.interceptor
+
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall
+import io.grpc.ForwardingServerCallListener.SimpleForwardingServerCallListener
+import io.grpc._
+import org.slf4j.Logger
+
+/** Adds basic logging around each gRPC call. */
+class LoggingServerInterceptor(logger: Logger) extends ServerInterceptor {
+
+  override def interceptCall[ReqT, RespT](
+      call: ServerCall[ReqT, RespT],
+      headers: Metadata,
+      next: ServerCallHandler[ReqT, RespT]
+  ): ServerCall.Listener[ReqT] = {
+    val methodName = call.getMethodDescriptor.getFullMethodName
+    val finalCall = new CloseServerCall(methodName, call)
+    new OnMessageServerCallListener(methodName, next.startCall(finalCall, headers))
+  }
+
+  private class CloseServerCall[A, B](methodName: String, delegate: ServerCall[A, B]) extends SimpleForwardingServerCall[A, B](delegate) {
+    override def close(status: Status, trailers: Metadata): Unit = {
+      if (!status.isOk) {
```

Comment:
In the original code, we have this:
```java
if (status.getCode() == Status.Code.UNKNOWN || status.getCode() == Status.Code.INTERNAL) {
  logger.error(String.format("Status code of response to be sent to client from method %s not OK. %s: %s", method.getFullMethodName(), status.getCode(), status.getDescription()), status.getCause());
}
else if (!status.isOk()) {
   logger.warn(String.format("Status code of response to be sent to client from method %s not OK. %s: %s", method.getFullMethodName(), status.getCode(), status.getDescription()), status.getCause());
}
else {
   logger.debug("Status code of response to be sent to client from method {} OK: {}", method.getFullMethodName(), status);
}
```
This is because we didn't want to log as `error` when a client sent invalid request (we log it as `warn` instead).

I'm not saying that the logic in the original code is perfect but we could this about these two categories of errors (client vs. server) and log them differently.

Reply:
Oh, I missed that there was different log level.